modules/libcom/src/iocsh/iocsh.h

```diff
@@ -10,6 +10,27 @@
 /* iocsh.h  ioc: call registered function*/
 /* Author:  Marty Kraimer Date: 27APR2000 */
 
+/**
+ * @file iocsh.h
+ * 
+ * @brief C and C++ defintions of functions for IOC shell programming.
+ * 
+ * @details
+ * The iocsh API provides an interface for running commands in the shell
+ * of the IOC, as well as registering commands and variables for use in the shell.
+ * It consists of 4 functions for the former and 2 functions for the latter.
+ * 
+ * @par Command functions:
+ *     int iocsh (const char *pathname)@n 
+ *     int iocshLoad (const char *pathname, const char *macros)@n 
+ *     int iocshCmd (const char *cmd)@n 
+ *     int iocshRun (const char *cmd, const char *macros)
+ * 
+ * @par Registration functions:
+ *     void iocshRegister (const iocshFuncDef * piocshFuncDef, iocshCallFunc func)@n 
+ *     void epicsStdCall iocshRegisterVariable (const iocshVarDef *piocshVarDef)
```

Comment:
I found Doxygen also supports markdown syntax, so you could write these lists with markdown list syntax and avoid the @n at the end.  No need to do that for this pull request, just FYI in case you're writing doxygen annotations in the future.

```suggestion
 * @par Command functions:
 *     - int iocsh (const char *pathname)
 *     - int iocshLoad (const char *pathname, const char *macros)
 *     - int iocshCmd (const char *cmd)
 *     - int iocshRun (const char *cmd, const char *macros)
 * 
 * @par Registration functions:
 *     void iocshRegister (const iocshFuncDef * piocshFuncDef, iocshCallFunc func)@n 
 *     void epicsStdCall iocshRegisterVariable (const iocshVarDef *piocshVarDef)
```